src/packages/core/components/field-dropdown-list/field-dropdown-list.element.ts

```diff
@@ -0,0 +1,213 @@
+
+	@state()
+	private _uniqueName?: string;
+
+	@state()
+	private _unique?: string;
+
+	@query('#value')
+	private _valueElement?: UUIComboboxElement;
+
+	#documentTypeDetailRepository = new UmbDocumentTypeDetailRepository(this);
+	#mediaTypeDetailRepository = new UmbMediaTypeDetailRepository(this);
+	#modalManager?: UmbModalManagerContext;
+
+	@state()
+	private _customFields: Array<Partial<UmbPropertyTypeModel>> = [];
+
+	private _systemFields: Array<Partial<UmbPropertyTypeModel>> = [
+		{ alias: 'sortOrder', name: this.localize.term('general_sort') },
+		{ alias: 'updateDate', name: this.localize.term('content_updateDate') },
+		{ alias: 'updater', name: this.localize.term('content_updatedBy') },
+		{ alias: 'createDate', name: this.localize.term('content_createDate') },
+		{ alias: 'owner', name: this.localize.term('content_createBy') },
+		{ alias: 'published', name: this.localize.term('content_isPublished') },
+		{ alias: 'contentTypeAlias', name: this.localize.term('content_documentType') },
+		{ alias: 'email', name: this.localize.term('general_email') },
+		{ alias: 'username', name: this.localize.term('general_username') },
+	];
+
+	constructor() {
+		super();
+		this.consumeContext(UMB_MODAL_MANAGER_CONTEXT, (modalManager) => {
+			this.#modalManager = modalManager;
+		});
+	}
+
+	async #getDocumentTypeFields() {
+		if (!this.#modalManager) return;
+		const modalContext = this.#modalManager.open(UMB_DOCUMENT_TYPE_PICKER_MODAL, {
+			data: {
+				hideTreeRoot: true,
+				multiple: false,
+			},
+		});
+
+		const modalValue = await modalContext.onSubmit();
+		const unique = modalValue.selection[0] ?? '';
+
+		const { data } = await this.#documentTypeDetailRepository.requestByUnique(unique);
+		if (!data) return;
+
+		this._unique = data.unique;
+		this._uniqueName = data.name;
+		this._customFields = data.properties;
+	}
+
+	async #getMediaTypeFields() {
+		if (!this.#modalManager) return;
+		const modalContext = this.#modalManager.open(UMB_MEDIA_TYPE_PICKER_MODAL, {
+			data: {
+				hideTreeRoot: true,
+				multiple: false,
+			},
+		});
+
+		const modalValue = await modalContext.onSubmit();
+		const unique = modalValue.selection[0] ?? '';
+
+		const { data } = await this.#mediaTypeDetailRepository.requestByUnique(unique);
+		if (!data) return;
+
+		this._unique = data.unique;
+		this._uniqueName = data.name;
+		this._customFields = data.properties;
+	}
+
+	#onChange(e: UUIComboboxEvent) {
+		this._type = (e.composedPath()[0] as UUIComboboxElement).value as FieldType;
+		this.value = undefined;
+		if (this._valueElement) this._valueElement.value = '';
+
+		switch (this._type) {
+			case FieldType.DOCUMENT_TYPE:
+				this.#getDocumentTypeFields();
+				break;
+			case FieldType.MEDIA_TYPE:
+				this.#getMediaTypeFields();
+				break;
+			default:
+				this._uniqueName = '';
+				this._unique = '';
+				this._customFields = this._systemFields;
+				this.value = undefined;
+				break;
+		}
+		this.dispatchEvent(new UmbChangeEvent());
```

Comment:
Minor suggestion, because we have multiple `this.dispatchEvent(new UmbChangeEvent());` calls (lines 141 and 149), which happens when the `this.value` is changed, we could combine them within a setter method for `value`.

Reply:
True, I implemented it :)